ndcube/ndcube.py

```diff
@@ -85,22 +87,30 @@ def world_axis_physical_types(self):
         pass
 
     @abc.abstractmethod
-    def crop_by_coords(self, min_coord_values, interval_widths):
+    def crop_by_coords(self, lower_corner, interval_widths=None, upper_corner=None):
```

Comment:
An API point here, I think it would be worth moving `interval_widths` into `**kwargs` so that it is no longer part of the function signature. We can still support it by doing a `kwargs.pop('interval_widths', None)` as the ~first line of the method.

Reply:
Hi, @DanRyanIrish told me to do it like that because some people will use this version as the previous : by inputting `corp_by_coords( (low1, low2), (inter1, inter2) )`. 

If I move `interval_widths` into `**kwargs`, I am afraid that people use `upper_corner` instead of `interval_widths` and the result will not be correct. 

I don't know if you saw it but I add a warning about using `interval_widths` where I ask the user to use `upper_corner`.

Of course I am open to every modification and if you want I change it, I'll do it !

Reply:
ah yes, you are correct. never mind then.